test/externalTests/perpetual-pools.sh

```diff
@@ -45,7 +45,7 @@ function perpetual_pools_test
         "${compile_only_presets[@]}"
         #ir-no-optimize            # Compilation fails with "YulException: Variable var_amount_527 is 9 slot(s) too deep inside the stack."
         #ir-optimize-evm-only      # Compilation fails with "YulException: Variable var_amount_527 is 9 slot(s) too deep inside the stack."
-        #ir-optimize-evm+yul       # Compilation fails with "YulException: Variable expr_mpos is 1 too deep in the stack"
+        ir-optimize-evm+yul       # Compilation fails with "YulException: Variable expr_mpos is 1 too deep in the stack"
```

Comment:
On a casual look, seems to compile but has testing issues https://app.circleci.com/pipelines/github/ethereum/solidity/21737/workflows/3588ee0e-de5a-4b29-81ce-fe17cff72fbb/jobs/952230

Reply:
This looks like https://github.com/nomiclabs/hardhat/issues/2115. I.e. a problem with Hardhat's heuristics not being compatible with IR yet. I got this in two other cases (OpenZeppelin and PRBMath) and the only thing we can do for now is to skip tests.

You can put it in the `compile_only_presets` array so that it at least gets compiled. Here's how I'm doing it for OZ: https://github.com/ethereum/solidity/blob/76540b338e0b354a1d55e33062c151a59fb67b50/test/externalTests/zeppelin.sh#L42-L44